### ISSUE TITLE
Weird rendering with the latest update

### ISSUE THREAD

**Issue by amodelbello** [entozoon/html-includes#9]
Getting stuff like this:
<img width="301" alt="screen shot 2018-02-26 at 10 31 49 pm" src="https://user-images.githubusercontent.com/2342804/36712349-9d09aee2-1b45-11e8-93dd-4bbeee369e31.png">
When it should look like this:
<img width="220" alt="screen shot 2018-02-26 at 10 32 10 pm" src="https://user-images.githubusercontent.com/2342804/36712364-a81f5ce6-1b45-11e8-90e1-0f6d0077e254.png">

**Comment by amodelbello** [comment]
https://github.com/entozoon/html-includes/pull/8 for this

**Comment by entozoon** [comment]
Hopefully resolved along with [#8](https://github.com/entozoon/html-includes/pull/8), so closing for now dude

**Comment by amodelbello** [comment]
Hmm. The issue is back for me with the 2.1.0 update. Dunno. Maybe I'm an edge case? I'll just use my fork in the meantime. What issues did the `xmlMode: false` cause?

**Comment by entozoon** [comment]
when xmlMode is false, if you try includes within `<head></head>` it whips them out and jams into `<body>` because it uses a jQuery parser thing. It's a [known issue](https://github.com/cheeriojs/cheerio/issues/1031) in fact :(

What's your source html there? I'll keep investigating

**Comment by amodelbello** [comment]
Okay I see what you mean about the `<head>` includes. I see a couple other issues with `xmlMode: true` however.

1. It tries to close `<link>` and `<meta>` tags in `<head>`
1. It adds closing `<body>` and `<html>` tags _before_ my main content and my _bottom.html include file. Fixing these two things however doesn't seem to fix the weird formatting I posted above but should probably be addressed in any event.

Here's some a formatted version of the output for my index.html file:
```<!DOCTYPE html>
<html lang="en">

<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <meta http-equiv="X-UA-Compatible" content="ie=edge">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css">
  <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css" integrity="sha384-/Y6pD6FV/Vv2HJnA6t+vslU6fwYXjCFtcEpHbNJ0lyAFsXTsjBbfaDjzALeQsN6M"
    crossorigin="anonymous">
  <style>
    /* Sticky footer styles */

    html {
      position: relative;
      min-height: 100%;
    }

    body {
      /* Margin bottom by footer height */
      margin-bottom: 60px;
    }

    .footer {
      position: absolute;
      bottom: 0;
      width: 100%;
      /* Set the fixed height of the footer here */
      height: 60px;
      line-height: 60px;
      /* Vertically center the text there */
      background-color: #f5f5f5;
    }
  </style>
  <title>WeightLister Practice</title>
  </link>
  </link>
  </meta>
  </meta>
  </meta>
</head>

<body>
  <header>
    <nav class="navbar navbar-expand-sm navbar-dark bg-dark mb-3">
      <div class="container">
        <a href="index.html" class="navbar-brand">WeightLister</a>
        <button class="navbar-toggler" data-toggle="collapse" data-target="#navbarNav">
          <span class="navbar-toggler-icon" />
        </button>
        <div class="collapse navbar-collapse" id="navbarNav">
          <ul class="navbar-nav">
            <li class="nav-item">
              <a href="workouts.html" class="nav-link">Workouts</a>
            </li>
            <li class="nav-item">
              <a href="exercises.html" class="nav-link">Exercises</a>
            </li>
          </ul>
          <div class="nav-item dropdown ml-sm-auto ml-xs-0">
            <a href="#" class="nav-link dropdown-toggle text-secondary pl-0" data-toggle="dropdown">Account &nbsp;&nbsp;
              <i class="fa fa-user" />
            </a>
            <div class="dropdown-menu">
              <a href="#" class="dropdown-item">My Account</a>
              <a href="#" class="dropdown-item">Preferences</a>
              <a href="#" class="dropdown-item">Logout</a>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>
</body>

</html>
<!-- ################ Begin Page Content ################### -->
<main role="main" class="container"> DASHBOARD </main>
<!-- ################ End Page Content ################### -->
<footer class="footer">
  <div class="container text-right">
    <span class="text-muted">&copy; 2018 ADB</span>
  </div>
</footer>
<script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN"
  crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.11.0/umd/popper.min.js" integrity="sha384-b/U6ypiBEHpOf/4+1nzFpr53nxSS+GLCkfwBdFNTxtclqqenISfwAzpKaMNFNmj4"
  crossorigin="anonymous" />
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/js/bootstrap.min.js" integrity="sha384-h0AbiXch4ZDo7tp9hKZ4TsHbi047NrKGLO3SEJAg45jXxnGIfYzk4Si90RDIqNm1"
  crossorigin="anonymous" />```

**Comment by entozoon** [comment]
Hmmm, that is quite significant of a problem. Bloody heck Cheerio!
Would you mind sending over the source files? (_header.html, _bottom.html or whatever)

It looks like perhaps rewriting with Cheerio was a bad choice. Bloody heck, I'll keep my ear to the ground for a new alternative, as hand parsing things with regex (as it did originally) was fraught with peril

**Comment by amodelbello** [comment]
Sure. here you go.
[src.zip](https://github.com/entozoon/html-includes/files/1799156/src.zip)

**Comment by entozoon** [comment]
Sorry about the delay, I've overhauled it with that hope that your issue is resolved with `v3.0.0`. Please open another issue if it persist, thanks!

(for future ref: cheerio was terrible and I've gone with a regex style inspired by webpack's html-loader)
